NDTensors/src/tupletools.jl

```diff
@@ -180,13 +180,13 @@ end
 
 Remove the value at pos and insert the elements in val
 """
-function insertat(t::NTuple{N}, val::NTuple{M}, pos::Integer) where {N,M}
+function insertat(t::Tuple, val::Tuple, pos::Integer)
+  N, M = length(t), length(val)
+  @assert pos <= N
```

Comment:
I put this assert because currently `pos == N+1` is allowed but leads to confusing, probably unintended behavior. In the `pos==N+1` case, this is the current behavior:
```
t = (1,2,3)
t = insertat(t,('a','b'),4)
@assert t == (1,2,'a')
```